collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.java

```diff
@@ -223,47 +216,10 @@ public void onLocationChanged(MapPoint point) {
     }
 
     /**
-     * Reacts to a tap on a feature by showing a toast or switching activities to view or edit a form.
+     * Reacts to a tap on a feature by showing a submission summary.
      */
     public void onFeatureClicked(int featureId) {
-        MappableFormInstance instance = instancesByFeatureId.get(featureId);
-        ClickAction clickAction = instance == null ? FormMapViewModel.ClickAction.NONE : instance.getClickAction();
-
-        boolean canEditSaved = (Boolean) AdminSharedPreferences.getInstance().get(AdminKeys.KEY_EDIT_SAVED);
-
-        switch (clickAction) {
-            case DELETED_TOAST:
-                String deletedTime = getString(R.string.deleted_on_date_at_time);
-                String disabledMessage = new SimpleDateFormat(deletedTime,
-                        Locale.getDefault()).format(viewModel.getDeletedDateOf(instance.getDatabaseId()));
-
-                ToastUtils.showLongToast(disabledMessage);
-                break;
-            case NOT_VIEWABLE_TOAST:
-                ToastUtils.showLongToast(R.string.cannot_edit_completed_form);
-                break;
-            case OPEN_READ_ONLY:
-                startActivity(getViewOnlyFormInstanceIntentFor(featureId));
-                break;
-            case OPEN_EDIT:
-                if (canEditSaved) {
-                    startActivity(getEditFormInstanceIntentFor(featureId));
-                } else {
-                    startActivity(getViewOnlyFormInstanceIntentFor(featureId));
-                }
-                break;
-        }
-    }
-
-    private Intent getEditFormInstanceIntentFor(int featureId) {
-        Uri uri = ContentUris.withAppendedId(InstanceColumns.CONTENT_URI, instancesByFeatureId.get(featureId).getDatabaseId());
-        return new Intent(Intent.ACTION_EDIT, uri);
-    }
-
-    private Intent getViewOnlyFormInstanceIntentFor(int featureId) {
-        Intent intent = getEditFormInstanceIntentFor(featureId);
-        intent.putExtra(ApplicationConstants.BundleKeys.FORM_MODE, ApplicationConstants.FormModes.VIEW_SENT);
-        return intent;
+        showSummary(featureId);
```

Comment:
It doesn't make sense to have both `onFeatureClicked` and `showSummary` since they have the same parameters and `showSummary` is only called from here. I'd keep `onFeatureClicked` and move the code here but you could also rename it to `showSummary` if you think that's clearer.

Reply:
Fixed.